### ISSUE TITLE
[tracking] running validator via CLI tools: cleanup

### ISSUE THREAD

**Issue by farazdagi** [prysmaticlabs/prysm#5659]
### Description

- We need to make sure that running CLI version of validator is easy, safe, and correctly described in docs.
- This issue is to track multiple minor (some even cosmetic) issues
- Targets all platforms and includes corresponding documentation updates (once code is updated, corresponding PR in docs will be created)

### Todo List

- [ ] generate deposit only after account keys are saved on disk #5552
- [ ] do not require to `--password`, allow specifying password interactively
  - [ ] `prysm.sh`
    - [x] `validator accounts create`
    - [ ] `validators accounts keys`
    - [ ] update documentation page (for OSX/Wins)
  - [ ] `prysm.bat` validator
    - [ ] `validator accounts create`
    - [ ] `validators accounts keys`
    - [ ] update documentation page (for Wins)
- [ ] default `keystore-path` value
  - [ ] make sure that it is not required (user can omit `keystore-path` and have sane defaults)
  - [ ] make sure the default value is consistent everywhere i.e. when omitted no errors occur
  - [ ] allow symlinks #5412 
  - [ ] `prysm.sh`
    - [x] `validator accounts create`
    - [ ] `validators accounts keys`
    - [ ] update documentation page (for OSX/Wins)
  - [ ] `prysm.bat` validator
    - [ ] `validator accounts create`
    - [ ] `validators accounts keys`
    - [ ] update documentation page (for Wins)
  - [ ] Decide on correct default paths:
    - [ ] After brief discussion on Discord the following concern has been raised: Usage of "Roaming" in path may result in private keys or large DB files getting copied around unintentionally. Use `AppData/Local/Eth2Validators` instead #5660
- [ ] Double check documentation (try running those examples)
  - [ ] Make sure that provided instructions are runnable (no errors, no missing deps) i.e. run them!
  - [x] https://docs.prylabs.network/docs/install/mac/activating-a-validator/ `prysm.sh -> ./prysm.sh`

**Comment by wgknowles** [comment]
https://github.com/prysmaticlabs/documentation/pull/61

**Comment by wgknowles** [comment]
Paths on Windows: https://github.com/prysmaticlabs/prysm/issues/5660

**Comment by har00ga** [comment]
This should probably have been opened on the docs repo, I didn't even see it until now :P No problem, all very quick fixes. Will do this aft.

**Comment by farazdagi** [comment]
@har00ga I was going through all the updates to the validator tool, and wanted to assign (or create PRs in docs) once I have this whole issue handled (to avoid many tiny fix requests). So, atm this tracker is merely for me not to miss anything *in code* - that's why I haven't assigned issue to you.

**Comment by terencechain** [comment]
@farazdagi Some background. @prestonvanloon and I have been working on consolidating validator key management here. Feel free to skip that part for the scope / or collaborate with us here:
https://docs.google.com/document/d/1u_P8veBtIj97u33y_bhOxmVOJJxkMHGNPoF6TvFhz2w/edit#heading=h.4d4hnezaletr

**Comment by wgknowles** [comment]
I can confirm this works:
prysm.bat validator accounts create
Also created prysmaticlabs/documentation#87 to remove references

validators accounts keys still requires --password flag

**Comment by 0xKiwi** [comment]
Can we also look into making deposit data reproducible? Currently it's just made once, and pretty much unretrievable since theres no way to connect withdrawal keys with the relevant sharding keys.

**Comment by wgknowles** [comment]
reated prysmaticlabs/documentation#95 to consistently use default paths and stop recommending putting cleartext passwords in a json file.

**Comment by rauljordan** [comment]
Closing in favor of https://github.com/prysmaticlabs/prysm/issues/6220
